examples/src/main/scala/io/finch/middleware/Main.scala

```diff
@@ -0,0 +1,58 @@
+package io.finch.middleware
+
+import cats.effect.IO
+import com.twitter.finagle.Http
+import com.twitter.finagle.http.{Response, Status}
+import com.twitter.util.{Await, Time}
+import io.finch._
+
+/**
+  * Small Finch hello world application serving endpoint protected by serious authentication
+  * where each request & response are also logged and measured.
+  *
+  * This is achieved using Kleisli-based middleware together with [[Bootstrap.compile]]
+  *
+  * Use the following curl commands to test it:
+  *
+  * {{{
+  *   curl -v -H "Authorization: secret" http://localhost:8081/hello
+  *   curl -V -H "Authorization: wrong" http://localhost:8081/hello
+  * }}}
+  */
+object Main extends App with Endpoint.Module[IO] {
```

Comment:
@sergeykolbasov I think you're on a right track with this. I looked at what http4s is doing and they pretty much follow the same structure: given a Kleilsi, wrap it with a bonus Klaisli. Take a look at these (https://github.com/http4s/http4s/tree/master/server/src/main/scala/org/http4s/server/middleware) for more inspiration.